diff --git a/lib/minify.js b/lib/minify.js
index 725e3d96..b256907c 100644
--- a/lib/minify.js
+++ b/lib/minify.js
@@ -103,6 +103,10 @@ function log_input(files, options, fs, debug_folder) {
 }
 
 async function minify(files, options, _fs_module) {
+    return minifySync(files, options, _fs_module);
+}
+
+function minifySync(files, options, _fs_module) {
     if (
         _fs_module
         && typeof process === "object"
@@ -370,5 +374,6 @@ async function minify(files, options, _fs_module) {
 
 export {
   minify,
+  minifySync,
   to_ascii,
 };
diff --git a/main.js b/main.js
index 0a10db5a..8eb0dc81 100644
--- a/main.js
+++ b/main.js
@@ -1,8 +1,8 @@
 import "./lib/transform.js";
 import "./lib/mozilla-ast.js";
-import { minify } from "./lib/minify.js";
+import { minify, minifySync } from "./lib/minify.js";
 
-export { minify } from "./lib/minify.js";
+export { minify, minifySync } from "./lib/minify.js";
 export { run_cli as _run_cli } from "./lib/cli.js";
 
 export async function _default_options() {
